### DISCUSSION TITLE
Support for Jepsen history output

### DISCUSSION THREAD

**Discussion by smccarthy788** [nosqlbench/nosqlbench#1100] [Ideas]
The Jepsen project has seen a major advancement in the form of [elle](https://github.com/jepsen-io/elle). In particular:
> In general, Elle checks real-world histories in a matter of seconds to minutes, rather than seconds to millennia. Where Knossos is often limited to a few hundred operations per history, Elle can handle hundreds of thousands of operations easily.

NoSqlBench provides a means to execute diverse workloads against multiple data systems. As such it is a good replacement for the history generation aspects of the Jepsen project. However it would be great to also be able to capitalize on the abilities of elle.

In order to do that NoSqlBench would need to be able to produce a jepsen history format output.

**Comment by jshook** [comment]
This should be somewhat easy to implement. I think it would be a great addition. However, I think the existing format seems quite fluffy in terms of IO bandwidth. For a performance-oriented tool like NB, I think we would want to have a compact logging form which could easily be expanded out to the jepsen history form. Introducing lots of disk IO is something we try to avoid when possible. In this case, it would be an essential part of a test (with the correctness-testing aspect) but we can make it efficient as well.

**Comment by dave2wave** [comment]
This seems to be oriented to running NB in a Fallout environment.
